src/dialects/abstract/data-types.ts

```diff
@@ -0,0 +1,1826 @@
+   * @param {string|number} [scale] defines scale
+   */
+  constructor(options?: DecimalOptions);
+  constructor(precision: number, scale?: number);
+  constructor(...args: DecimalConstructorParams);
+  constructor(precision?: number | DecimalOptions, scale?: number) {
+    if (typeof precision === 'object') {
+      super(precision);
+      this.options ??= precision;
+    } else {
+      super();
+      // Sadly TS isn't smart enough to infer that this will be defined by the constructor of NUMBER.
+      // This should never do anything becuase this.options should already be defined.
+      this.options ??= {};
+
+      this.options.precision = precision;
+      this.options.scale = scale;
+    }
+
+    this._precision = this.options.precision;
+    this._scale = this.options.scale;
+  }
+
+  public toSql() {
+    if (this._precision || this._scale) {
+      return `DECIMAL(${[this._precision, this._scale]
+        .filter(num => num != null)
+        .join(',')})`;
+    }
+
+    return 'DECIMAL';
+  }
+
+  public validate(value: AcceptedNumber): asserts value is AcceptedNumber;
+  public validate(value: AcceptedNumber): true {
+    if (!Validator.isDecimal(String(value))) {
+      throw new ValidationError(
+        util.format('%j is not a valid decimal', value),
+        [],
+      );
+    }
+
+    return true;
+  }
+}
+
+/**
+ * Decimal type, variable precision, take length as specified by user
+ */
+export type DECIMAL = _DECIMAL;
+export const DECIMAL = classToInvokable(_DECIMAL);
+
+/**
+ * A boolean / tinyint column, depending on dialect
+ */
+class _BOOLEAN extends ABSTRACT<
+  boolean | Falsy,
+  boolean,
+  string | number | Buffer | boolean
+> {
+  public static readonly key = 'BOOLEAN';
+
+  public toSql() {
+    // Note: This may vary depending on the dialect.
+    return 'TINYINT(1)';
+  }
+
+  public validate(value: boolean | Falsy): asserts value is boolean;
+  public validate(value: boolean | Falsy): true {
+    if (!Validator.isBoolean(String(value))) {
+      throw new ValidationError(
+        util.format('%j is not a valid boolean', value),
+        [],
+      );
+    }
+
+    return true;
+  }
+
+  protected _sanitize(
+    value: string | number | Buffer | boolean,
+  ): boolean | null {
+    if (value !== null && value !== undefined) {
+      if (Buffer.isBuffer(value) && value.length === 1) {
+        // Bit fields are returned as buffers
+        value = value[0];
+      }
+
+      const type = typeof value;
+      if (type === 'string') {
+        // Only take action on valid boolean strings.
+        if (value === 'true') {
+          return true;
+        }
+
+        if (value === 'false') {
+          return false;
+        }
+      } else if (
+        type === 'number' // Only take action on valid boolean integers.
+        && (value === 0 || value === 1)
+      ) {
+        return Boolean(value);
+      }
+    }
+
+    return null;
+  }
+
+  static readonly parse = this.prototype._sanitize;
```

Comment:
I'd prefer to have the `_sanitize` instance method call the static `parse` method instead of getting something from the prototype